src/js/components/AppListComponent.jsx

```diff
@@ -188,7 +189,42 @@ var AppListComponent = React.createClass({
     });
   },
 
-  filterNodes: function (nodesSequence, filterCounts) {
+  getGroupsFromApps: function (apps) {
+    var groups = [];
+
+    apps.forEach(app => {
+      let pathParts = app.id.split("/");
+
+      // Remove app name
+      pathParts.pop();
+
+      // Create groups
+      pathParts.forEach((pathPart, index, pathParts) => {
+        // We can ignore the first part as all app/group ids begin with a slash
+        if (index === 0) {
+          return;
+        }
+
+        let groupId = pathParts.slice(0, index + 1).join("/");
+        let group = groups.find(item => {
+          return item.id === groupId;
+        });
+
+        if (group == null) {
+          group = initGroupNode(groupId, app);
+          groups.push(group);
+        } else {
+          updateGroupNode(group, app);
+        }
+      });
+
+    });
+
+    return groups;
+  },
```

Comment:
Let's do this refactoring in another PR along with the other places where we work with the app path, as requested in mesosphere/marathon#3197